Created CMM Simulink Block

CarlottaSartore opened the pull request:
As commented in issue #20 , This PR contains the implementation of the Simulink block calling the `iDynTree::KinDynComputations::getCentroidalTotalMomentumJacobian()` function implemented by @GiulioRomualdi .

C.C. @traversaro @gabrielenava

CarlottaSartore commented:
I saw that the CI failed, I have tried again the compilation from a brand new build and everything worked fine. 

I checked the output of the CI tests and I saw that the problem is in the iDynTree dependency i.e. 

```
../toolbox/library/src/CMM.cpp:176:18: error: 'using element_type = class iDynTree::KinDynComputations {aka class iDynTree::KinDynComputations}' has no member named 'getCentroidalTotalMomentumJacobian'; did you mean 'getCentroidalTotalMomentum'?
     ok = kinDyn->getCentroidalTotalMomentumJacobian(pImpl->CentroidalTotalMomentumMatrix);
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                  getCentroidalTotalMomentum
[26/30] Building CXX object toolbox/library/CMakeFiles/WBToolbox.dir/src/DiscreteFilter.cpp.o
```

Which branch is the CI using ? is it possible that the error is due to  "wrong" iDynTree version? 

In my configuration I have iDynTree branch devel commit 03a667417dcc4f5ac2ec881bad99cc24cea24947.

diegoferigo commented:
I think that the images in https://github.com/robotology-playground/docker_images should be updated, if we're lucky we just need to re-run its CI. Let's wait and then retrigger the tests of this PR.

diegoferigo commented on toolbox/library/CMakeLists.txt:
```diff
@@ -38,6 +38,7 @@ list(APPEND WBT_BLOCKS "Jacobian")
 list(APPEND WBT_BLOCKS "ForwardKinematics")
 list(APPEND WBT_BLOCKS "RelativeTransform")
 list(APPEND WBT_BLOCKS "CentroidalMomentum")
+list(APPEND WBT_BLOCKS "CMM")
```
I would extend CMM to CentroidalTotalMomentumMatrix pretty much everywhere (class names, files names, etc)

CarlottaSartore replied:
Perfect, I will apply this change right away!

CarlottaSartore replied:
Done in commit 95fa7bcb51c3f942271b80777c4861f7eb814e3c

diegoferigo commented on toolbox/library/src/CMM.cpp:
```diff
@@ -0,0 +1,202 @@
+{}
+
+CMM::~CMM() = default;
+
+
+
```
I don't think that the clang-format of this repo allows multiple blank lines. Are you sure you're using it? If not, can you update the style of all the C++ file with that?

CarlottaSartore replied:
Yes sure !

CarlottaSartore replied:
Done in commit 95fa7bcb51c3f942271b80777c4861f7eb814e3c

CarlottaSartore commented:
@diegoferigo  Thank you for your review! 

>  did you export the mdl and slx for the correct Matlab version we support?

For exporting the slx I have used the [export_library](https://github.com/robotology/wb-toolbox/blob/devel/matlab/export_library.m) script, instead the mdl version is 2019b, which Matlab version is the default one? 2014b ?

traversaro commented:
Just to understand, @CarlottaSartore you have addressed all the reviews of @diegoferigo, right?

CarlottaSartore commented:
> Just to understand, @CarlottaSartore you have addressed all the reviews of @diegoferigo, right?

Yes, the only thing missing, to the best of my knowledge,  regards the version of Matlab as per the following comment: 

> For exporting the slx I have used the export_library script (i.e. it is 2014b), instead, the mdl version is 2019b, as Matlab version is the default one? 2014b ?

The CI tests are not passing but it could due to https://github.com/robotology/wb-toolbox/pull/194#issuecomment-660163374

diegoferigo commented:
> For exporting the slx I have used the export_library script (i.e. it is 2014b), instead, the mdl version is 2019b, which Matlab version is the default one? 2014b ?

I think that executing the `export_library` CMake should suffice, it already takes care of using the right version of Matlab:

https://github.com/robotology/wb-toolbox/blob/3fd8eee7835e54a0497e6bd8a72a3307a244124d/matlab/CMakeLists.txt#L9-L14

For what concerns CI, I triggered the generation of the updated Docker images, let's see if they succeed. The pipeline is a bit rusty lately, I had not much time to dedicate and there was no one else that was maintaining it. Let's see if the building process succeeds, it's gonna take a while.

diegoferigo commented:
Can you please rebase this PR on top of current `devel`? CI should be fixed now.

CarlottaSartore commented:
@diegoferigo  I have updated all the file with the clang-format (using QT), I have rebase on top of devel and fix one leftover. of the reanming in the simulink library. 
The version of the `WBToolboxLibrary.slx `(inside the exported folder) is R2014b and has been created using the `export_library` script.

diegoferigo commented:
Perfect, thanks for your contribution @CarlottaSartore, it looks good! A last nit, can you please squash together the two commits with the slx file? Minimizing the changes to binary files is very important to avoid the size of the git project to diverge :)
